### DISCUSSION TITLE
Resolving conflict between what topics should and should not take place in discussions.

### DISCUSSION THREAD

**Discussion by csmir** [Pryaxis/TShock#2466] [Ideas]
Following up on #2465 by request of @hakusaro. 

#### Quick recap:

![intro](https://rozen.one/files/msedge_Qz7oUPh0Zz.png)

**Comment by csmir** [comment]
The main issue that needs to be addressed here is that when posting plugins, a specific security & validity requirement stands. Showing off plugins in the discussion tab does not guarantee that any of the plugins are actually safe for use, nor can this be properly reviewed beforehand because there are no blockades in creating discussions.

Next to this, plugin discussion should take place at the repository where the plugin originally comes from or the current maintainers' repository. Plugin discussion here will not always be seen/acknowledged by the plugin developer unless its directly initiated by them, which is by all means poor initiative. If there are issues or matters to be discussed regarding a plugin, the developer(s) should take the lead in actually resolving it.

Lastly, discussions are a place to discuss matters that involve the library, even under the 'show and tell' tag. If you want to take discussion off-topic, the Discord is a place to do so. While this is an opinion, I do believe GitHub is not a place for free off-topic discussion, as it was not created for this purpose and does not serve as it (by default) now.

**Reply by moisterrific** [reply]
I agree, especially on the security part. All submissions in Pryaxis/Plugins are vetted so people can feel safe knowing they're not downloading malware. I think it sets a dangerous precedent if people start linking plugins that have not been thoroughly checked. 

Although @COMDARR haven't linked any `.dll` files directly, he does have them under releases in the respective repositories, and there's no assurance the binary has exactly the same code as the public repo. Not that I'm accusing him specifically of doing anything malicious, just saying that's a possibility in general if this becomes the norm.

**Reply by Arthri** [reply]
I agree, plugin release or update discussions are a problem, as both of you have mentioned, it doesn't go through the review process and might potentially be dangerous

However, I don't really mind plugin or project show-offs keeping in the spirit of show and tell. TShock has a general notion of being TShockAPI, TSAPI, and the community as a whole, so plugin show offs and that stuff aren't too far out of the picture

**Comment by bartico6** [comment]
why is that even a conversation we're having, isn't the plugins repo **literally made for that purpose**?

if the author made a new plugin they should post about it there, not on the main repo.

**Reply by hakusaro** [reply]
Did you even read what I said?

**Comment by comdar16** [comment]
literally posted some updated plugins and everyone started crying about it.

**Reply by csmir** [reply]
Nobody cried about anything. A healthy discussion was started about if plugins belong in the main repo with valid points presented why they shouldn't. This reply is childish.

**Reply by comdar16** [reply]
> Nobody cried about anything. A healthy discussion was started about if plugins belong in the main repo with valid points presented why they shouldn't. This reply is childish.

if you can't put some updated plugins to tshock discussion then why category Show and tell exist

**Reply by bartico6** [reply]
Those are default Github categories, I'm pretty sure.

**Reply by csmir** [reply]
I can verify that, yes. Also would like to note that I never attacked you for creating any discussion because it's not noted that you shouldn't. I only advised you to take your plugins elsewhere as there are better options available. You still haven't done that, even while pushing to the plugins repository will have your plugin(s) get more acknowledgement.

**Reply by comdar16** [reply]
if i have options then i will leave all my discussions where they are now. If you want you can delete or report them i don't really care.

(Also i m new to github so i don't understand a single shit in repos / discussions e.t.c

**Reply by moisterrific** [reply]
> 
> 
> if i have options then i will leave all my discussions where they are now. If you want you can delete or report them i don't really care.
> 
> (Also i m new to github so i don't understand a single shit in repos / discussions e.t.c

thanks for updating those old plugins but you have to understand there's a process for everything. the proper procedure is to follow instructions on Pryaxis/Plugins and submit a pull request to the readme file with your updated plugin links, after doing so the plugin binary (.dll file) will be decompiled by pryaxis and checked for compliance and security, then finally the pull request is accepted once that's all clear. 

you have to understand from a tshock user's perspective, if they see a plugin someone linked in discussions they're going to think that's safe to download and use, but obviously that's not the case since it has not been reviewed, so nobody knows what the plugin .dll actually does except the person who published the release.

**Reply by comdar16** [reply]
i made these plugins open source you can literally view the code or make your own build

**Reply by hakusaro** [reply]
@COMDARR thank you! I sincerely apologize for the confusion here. I promise you that the point of me asking for this topic is to discuss the future of the plugin system/scheme. We're going to reach a good conclusion. I don't think you're a bad person, and I appreciate your work so far. Thank you!

Note: There is no actual stated process that says you can't post plugins on the show and tell section of this forum. While we offer a plugins repo that has a list of plugins, there isn't a law against distributing plugins in other ways, except on Discord, where we have to worry about the Discord ToS.

The actual point of me requesting this discussion was to get feedback from other plugin authors and plugin users on what they want to happen and what they expect to happen. My point is not to let people attack you. If you feel like someone is attacking you, please send me an email at argo at hey dot com, and I'll take care of kicking whoever is attacking you in the face. I do not condone ill-treatment of people who do nice things. Thank you for actually caring about plugins and working on them! 💚
